src/main/scala/io/iohk/ethereum/consensus/ethash/validators/OmmersValidator.scala

```diff
@@ -36,9 +37,10 @@ object OmmersValidator {
 
   object OmmersError {
     case object OmmersLengthError extends OmmersError
-    case object OmmersNotValidError extends OmmersError
+    case class OmmersHeaderError(error: Option[BlockHeaderError]) extends OmmersError
```

Comment:
An optional argument to an error class seems a bit weird to me as it's so uninformative. (Also commented below about this.)

Reply:
The internal BlockHeaderError was being completely discarded and replaced by a generic `OmmersNotValidError`. This error happens when during the ommers header validation and it can fail because of 18 different reasons, and that will be the specific BlockHeaderError.
It's an `Option` for code reasons, programmatically has to cover all possible cases.
This may indeed a code smell, why is there a test that returns a generic error running this code? I did not check to be honest.